corehq/tabs/tabclasses.py

```diff
@@ -59,9 +64,42 @@ def sidebar_items(self):
             request=self._request, domain=self.domain)
         custom_reports = CustomProjectReportDispatcher.navigation_sections(
             request=self._request, domain=self.domain)
-        sidebar_items = tools + report_builder_nav + custom_reports + project_reports
+        sidebar_items = tools + report_builder_nav + self._regroup_sidebar_items(custom_reports + project_reports)
         return self._filter_sidebar_items(sidebar_items)
 
+    def _regroup_sidebar_items(self, sidebar_items):
+        try:
+            ordering = ReportsSidebarOrdering.objects.get(domain=self.domain)
+        except ObjectDoesNotExist:
+            return sidebar_items
+
+        reports_to_move = {}  # report class name to SidebarPosition mapping
```

Comment:
it might be a little clearer to extract everything here and below to a utility function and replace this line with:
```python
return _reorder_sidebar_items(reports, ordering)
```
not sure if that would also simplify the testing interface at all, but seems like it might.